src/p2p/net_node.inl

```diff
@@ -332,6 +353,13 @@ namespace nodetool
       for (const auto &c: conns)
         zone.second.m_net_server.get_config_object().close(c);
 
+      for (int i = 0; i < 2; ++i)
+        zone.second.m_peerlist.filter(i == 0, [&subnet](const peerlist_entry &pe){
```

Comment:
How big do these lists get? Can we do a `2*log(n)` range search then erase? This should be possible because the iteration key is sorted. The one below is going to be more complicated though.

Reply:
https://github.com/monero-project/monero/blob/ed506006d2563c474e52db5a91ba2e20af86f30f/src/cryptonote_config.h#L130-L131

Reply:
Subnets are only banned at startup atm anyway. I doubt it matters much in practice.